iOS/Data/Actor/Observers/Realm+LibraryGrid.swift

```diff
@@ -14,14 +14,36 @@ struct LibraryGridState {
     let query: String
     let sort: LibraryView.LibraryGrid.KeyPath
     let order: LibraryView.LibraryGrid.SortOrder
-    let showOnlyDownloadedTitles: Bool
+    let filterByDownloadedTitles: Bool
+    let filterByTrackedTitles: Bool
+    let titlePinningType: TitlePinningType?
+
+    func getTitlePinningType() -> TitlePinningType? {
+        // Collection settings takes priority
+        if let collection, collection.pinningType != nil && collection.pinningType != TitlePinningType.none {
+            return collection.pinningType
+        }
+
+        // Global setting
+        if let titlePinningType, titlePinningType != TitlePinningType.none {
+            return titlePinningType
+        }
+
+        return nil
+    }
 }
 
 extension RealmActor {
-    func observeLibrary(state: LibraryGridState, _ callback: @escaping Callback<[LibraryEntry]>) async -> NotificationToken {
+
+    func getLibraryEntries(state: LibraryGridState) -> Results<LibraryEntry> {
+        let tracked = realm
```

Comment:
Sources that provide tracking endpoints for auto tracking will be filtered out here